exonum/src/runtime/rust/mod.rs

```diff
@@ -37,12 +36,25 @@ use crate::messages::{BinaryForm, CallInfo};
 use crate::proto::schema;
 use crate::storage::{Error as StorageError, Fork, Snapshot};
 
-use self::service::Service;
+use self::service::{Service, ServiceFactory};
+use crate::api::ServiceApiBuilder;
+use crate::runtime::configuration_new::{DEPLOY_METHOD_ID, INIT_METHOD_ID, SERVICE_ID};
+use crate::runtime::dispatcher::Dispatcher;
+use protobuf::well_known_types::Any as PbAny;
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct RustRuntime {
-    // TODO: think about ways to share runtime.
-    inner: RefCell<RustRuntimeInner>,
+    inner: RustRuntimeInner,
+    dispatcher: *mut Dispatcher,
```

Comment:
IMHO in the future we should rework all that pointers and unsafe code... It just smells bad.

Reply:
Yes, but how this should be implemented?
I tried using Weak-Rc. (Blockchain has Rc on Dispatcher), Config service has Weak on Dispatcher. 
But then when we execute Init transaction we have borrow on Dispatcher (since all transactions are calls to Dispatcher method) and we have to init service which will require mutable borrow on dispatcher. That is why I think all safe primitives such as RefCell and Rc can't be used here.
Maybe I am missing something.

Reply:
Currently I have no answer, that's a topic for further investigation.
Maybe one day we'll start using actor approach and then it'll be separate actors that interact asynchonously. Or we'll find other solution.